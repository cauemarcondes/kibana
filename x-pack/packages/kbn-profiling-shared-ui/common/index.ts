/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { i18n } from '@kbn/i18n';

export const NOT_AVAILABLE_LABEL = i18n.translate('xpack.profiling.notAvailableLabel', {
  defaultMessage: 'N/A',
});

export enum AddDataTabs {
  Kubernetes = 'kubernetes',
  Docker = 'docker',
  Binary = 'binary',
  Deb = 'deb',
  RPM = 'rpm',
  ElasticAgentIntegration = 'elasticAgentIntegration',
  Symbols = 'symbols',
}

export const PROFILING_FEEDBACK_LINK = 'https://ela.st/profiling-feedback';
