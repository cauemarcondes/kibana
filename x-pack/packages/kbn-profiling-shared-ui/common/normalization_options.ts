/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

export interface NormalizationOptions {
  baselineScale: number;
  baselineTime: number;
  comparisonScale: number;
  comparisonTime: number;
}

export enum ComparisonMode {
  Absolute = 'absolute',
  Relative = 'relative',
}

export enum NormalizationMode {
  Scale = 'scale',
  Time = 'time',
}
