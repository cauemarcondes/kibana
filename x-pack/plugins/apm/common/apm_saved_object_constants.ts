/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

// the types have to match the names of the saved object mappings
// in /x-pack/plugins/apm/mappings.json

// APM indices (legacy)
/**
 * Should use APM_INDICES_SPACE_SAVED_OBJECT_TYPE instead
 * @deprecated
 * */
export const APM_INDICES_SAVED_OBJECT_TYPE = 'apm-indices';
/**
 * Should use APM_INDICES_SPACE_SAVED_OBJECT_ID instead
 * @deprecated
 * */
export const APM_INDICES_SAVED_OBJECT_ID = 'apm-indices';

// APM indices (space aware)
export const APM_INDICES_SPACE_SAVED_OBJECT_TYPE = 'apm-indices-space';
export const APM_INDICES_SPACE_SAVED_OBJECT_ID = 'apm-indices-space';

// APM telemetry
export const APM_TELEMETRY_SAVED_OBJECT_TYPE = 'apm-telemetry';
export const APM_TELEMETRY_SAVED_OBJECT_ID = 'apm-telemetry';

// APM Server schema
export const APM_SERVER_SCHEMA_SAVED_OBJECT_TYPE = 'apm-server-schema';
export const APM_SERVER_SCHEMA_SAVED_OBJECT_ID = 'apm-server-schema';
