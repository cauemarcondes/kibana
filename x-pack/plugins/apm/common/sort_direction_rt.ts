/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import * as t from 'io-ts';

export const toSortDirectionRt = t.union([t.literal('asc'), t.literal('desc')]);
export type SortDirection = t.TypeOf<typeof toSortDirectionRt>;
