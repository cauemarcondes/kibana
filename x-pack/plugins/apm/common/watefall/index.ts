/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { ErrorRaw } from '../../typings/es_schemas/raw/error_raw';
import { TimestampUs } from '../../typings/es_schemas/raw/fields/timestamp_us';

export interface WaterfallErrorDoc {
  timestamp: TimestampUs;
  trace: { id?: string };
  transaction: {
    id?: string;
  };
  parent: { id?: string };
  error: ErrorRaw['error'];
  service: {
    name: string;
  };
}
