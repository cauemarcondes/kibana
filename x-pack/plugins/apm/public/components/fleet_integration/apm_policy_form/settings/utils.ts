/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { PackagePolicyValues } from '../typings';

function mergeNewValue(
  oldValues: PackagePolicyValues,
  key: string,
  value?: any
): PackagePolicyValues {
  return { ...oldValues, [key]: { ...oldValues[key], value } };
}

export function handleFormChange({
  values,
  key,
  value,
  validateForm,
}: {
  values: PackagePolicyValues;
  key: string;
  value?: any;
  validateForm: (values: PackagePolicyValues) => boolean;
}) {
  const newValues = mergeNewValue(values, key, value);
  const isValid = validateForm(newValues);
  return { newValues, isValid };
}
