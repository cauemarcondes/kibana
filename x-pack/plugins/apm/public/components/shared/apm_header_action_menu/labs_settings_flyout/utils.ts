/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { toEditableConfig } from '@kbn/advanced-settings-plugin/public';
import { IUiSettingsClient } from '@kbn/core/public';

type EditableConfig = Record<string, ReturnType<typeof toEditableConfig>>;

export function getEditableConfig({
  settingsKeys,
  uiSettings,
}: {
  settingsKeys: string[];
  uiSettings?: IUiSettingsClient;
}) {
  if (!uiSettings) {
    return {};
  }
  const uiSettingsDefinition = uiSettings.getAll();
  const config: EditableConfig = {};

  settingsKeys.forEach((key) => {
    const settingDef = uiSettingsDefinition?.[key];
    if (settingDef) {
      const editableConfig = toEditableConfig({
        def: settingDef,
        name: key,
        value: settingDef.userValue,
        isCustom: uiSettings.isCustom(key),
        isOverridden: uiSettings.isOverridden(key),
      });
      config[key] = editableConfig;
    }
  });
  return config;
}
