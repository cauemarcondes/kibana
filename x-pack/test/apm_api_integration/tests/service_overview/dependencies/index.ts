/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import expect from '@kbn/expect';
import { last, pick, sortBy } from 'lodash';
import { ValuesType } from 'utility-types';
import moment from 'moment';
import { createApmApiSupertest } from '../../../common/apm_api_supertest';
import { roundNumber } from '../../../utils';
import { ENVIRONMENT_ALL } from '../../../../../plugins/apm/common/environment_filter_values';
import { APIReturnType } from '../../../../../plugins/apm/public/services/rest/createCallApmApi';
import archives from '../../../common/fixtures/es_archiver/archives_metadata';
import { FtrProviderContext } from '../../../common/ftr_provider_context';
import { registry } from '../../../common/registry';
import { apmDependenciesMapping, createServiceDependencyDocs } from './es_utils';

export default function ApiTest({ getService }: FtrProviderContext) {
  const apmApiSupertest = createApmApiSupertest(getService('supertest'));
  const es = getService('es');

  const archiveName = 'apm_8.0.0';
  const timeRange = archives[archiveName];
  const comparisonStart = timeRange.start;
  const comparisonEnd = moment(timeRange.start).add(15, 'minutes').toISOString();

  const start = moment(timeRange.end).subtract(15, 'minutes').toISOString();
  const end = timeRange.end;

  registry.when(
    'Service overview dependencies when data is not loaded',
    { config: 'basic', archives: [] },
    () => {
      it('handles the empty state', async () => {
        const response = await apmApiSupertest({
          endpoint: `GET /api/apm/services/{serviceName}/dependencies`,
          params: {
            path: { serviceName: 'opbeans-java' },
            query: {
              start,
              end,
              numBuckets: 20,
              environment: ENVIRONMENT_ALL.value,
              comparisonStart,
              comparisonEnd,
            },
          },
        });

        expect(response.status).to.be(200);
        expect(response.body).to.eql({ currentPeriod: [], previousPeriod: {} });
      });
    }
  );

  registry.when(
    'Service overview dependencies when specific data is loaded',
    { config: 'basic', archives: [] },
    () => {
      let response: {
        status: number;
        body: APIReturnType<'GET /api/apm/services/{serviceName}/dependencies'>;
      };

      const indices = {
        metric: 'apm-dependencies-metric',
        transaction: 'apm-dependencies-transaction',
        span: 'apm-dependencies-span',
      };

      const startTime = new Date(start).getTime();
      const endTime = new Date(end).getTime();

      after(async () => {
        const allIndices = Object.values(indices).join(',');
        const indexExists = (await es.indices.exists({ index: allIndices })).body;
        if (indexExists) {
          await es.indices.delete({
            index: allIndices,
          });
        }
      });

      before(async () => {
        await es.indices.create({
          index: indices.metric,
          body: {
            mappings: apmDependenciesMapping,
          },
        });

        await es.indices.create({
          index: indices.transaction,
          body: {
            mappings: apmDependenciesMapping,
          },
        });

        await es.indices.create({
          index: indices.span,
          body: {
            mappings: apmDependenciesMapping,
          },
        });

        const docs = [
          ...createServiceDependencyDocs({
            service: {
              name: 'opbeans-java',
              environment: 'production',
            },
            agentName: 'java',
            span: {
              type: 'external',
              subtype: 'http',
            },
            resource: 'opbeans-node:3000',
            outcome: 'success',
            responseTime: {
              count: 2,
              sum: 10,
            },
            time: startTime,
            to: {
              service: {
                name: 'opbeans-node',
              },
              agentName: 'nodejs',
            },
          }),
          ...createServiceDependencyDocs({
            service: {
              name: 'opbeans-java',
              environment: 'production',
            },
            agentName: 'java',
            span: {
              type: 'external',
              subtype: 'http',
            },
            resource: 'opbeans-node:3000',
            outcome: 'failure',
            responseTime: {
              count: 1,
              sum: 10,
            },
            time: startTime,
          }),
          ...createServiceDependencyDocs({
            service: {
              name: 'opbeans-java',
              environment: 'production',
            },
            agentName: 'java',
            span: {
              type: 'external',
              subtype: 'http',
            },
            resource: 'postgres',
            outcome: 'success',
            responseTime: {
              count: 1,
              sum: 3,
            },
            time: startTime,
          }),
          ...createServiceDependencyDocs({
            service: {
              name: 'opbeans-java',
              environment: 'production',
            },
            agentName: 'java',
            span: {
              type: 'external',
              subtype: 'http',
            },
            resource: 'opbeans-node-via-proxy',
            outcome: 'success',
            responseTime: {
              count: 1,
              sum: 1,
            },
            time: endTime - 1,
            to: {
              service: {
                name: 'opbeans-node',
              },
              agentName: 'nodejs',
            },
          }),
        ];

        const bulkActions = docs.reduce(
          (prev, doc) => {
            return [...prev, { index: { _index: indices[doc.processor.event] } }, doc];
          },
          [] as Array<
            | {
                index: {
                  _index: string;
                };
              }
            | ValuesType<typeof docs>
          >
        );

        await es.bulk({
          body: bulkActions,
          refresh: 'wait_for',
        });

        response = await apmApiSupertest({
          endpoint: `GET /api/apm/services/{serviceName}/dependencies`,
          params: {
            path: { serviceName: 'opbeans-java' },
            query: {
              start,
              end,
              numBuckets: 20,
              environment: ENVIRONMENT_ALL.value,
              comparisonStart,
              comparisonEnd,
            },
          },
        });
      });

      it('returns a 200', () => {
        expect(response.status).to.be(200);
      });

      it('returns dependencies on current period', () => {
        expect(response.body.currentPeriod.length).to.be(2);
      });

      it('returns dependencies on previous period', () => {
        expect(Object.keys(response.body.previousPeriod).length).to.be(2);
      });

      it('returns opbeans-node as a dependency', () => {
        const opbeansNode = response.body.currentPeriod.find(
          (item) => item.type === 'service' && item.serviceName === 'opbeans-node'
        );

        expect(opbeansNode !== undefined).to.be(true);

        expectSnapshot(opbeansNode).toMatch();

        const currentPeriodFirstValue = roundNumber(opbeansNode?.latency.timeseries[0].y);
        const currentPeriodLastValue = roundNumber(last(opbeansNode?.latency.timeseries)?.y);

        expectSnapshot(currentPeriodFirstValue).toMatchInline(`"6.667"`);
        expectSnapshot(currentPeriodLastValue).toMatchInline(`"1.000"`);
      });

      it('returns postgres as an external dependency', () => {
        const postgresCurrentPeriod = response.body.currentPeriod.find(
          (item) => item.type === 'external' && item.name === 'postgres'
        );

        const postgresPreviousPeriod = response.body.previousPeriod.postgres;

        expect(postgresCurrentPeriod !== undefined).to.be(true);
        expect(postgresPreviousPeriod !== undefined).to.be(true);

        const fields = ['spanType', 'spanSubtype', 'name', 'type'];
        const currentValues = {
          latency: roundNumber(postgresCurrentPeriod?.latency.value),
          throughput: roundNumber(postgresCurrentPeriod?.throughput.value),
          errorRate: roundNumber(postgresCurrentPeriod?.errorRate.value),
          impact: postgresCurrentPeriod?.impact,
          ...pick(postgresCurrentPeriod, ...fields),
        };

        expectSnapshot(currentValues).toMatchInline(`
          Object {
            "errorRate": "0.000",
            "impact": 0,
            "latency": "3.000",
            "name": "postgres",
            "spanSubtype": "http",
            "spanType": "external",
            "throughput": "0.06667",
            "type": "external",
          }
        `);

        const previousValues = {
          latency: roundNumber(postgresPreviousPeriod?.latency.value),
          throughput: roundNumber(postgresPreviousPeriod?.throughput.value),
          errorRate: roundNumber(postgresPreviousPeriod?.errorRate.value),
          impact: postgresPreviousPeriod?.impact,
          ...pick(postgresPreviousPeriod, ...fields),
        };

        expectSnapshot(previousValues).toMatchInline(`
          Object {
            "errorRate": "0.000",
            "impact": 0,
            "latency": "3.000",
            "name": "postgres",
            "spanSubtype": "http",
            "spanType": "external",
            "throughput": "0.06667",
            "type": "external",
          }
        `);
      });
    }
  );

  registry.when(
    'Service overview dependencies when data is loaded',
    { config: 'basic', archives: [archiveName] },
    () => {
      let response: {
        status: number;
        body: APIReturnType<'GET /api/apm/services/{serviceName}/dependencies'>;
      };

      before(async () => {
        response = await apmApiSupertest({
          endpoint: `GET /api/apm/services/{serviceName}/dependencies`,
          params: {
            path: { serviceName: 'opbeans-python' },
            query: {
              start,
              end,
              numBuckets: 20,
              environment: ENVIRONMENT_ALL.value,
              comparisonStart,
              comparisonEnd,
            },
          },
        });
      });

      it('returns a successful response', () => {
        expect(response.status).to.be(200);
      });

      it('returns at least one item', () => {
        expect(response.body.currentPeriod.length).to.be.greaterThan(0);
        expectSnapshot(response.body.currentPeriod[0]).toMatch();
      });

      it('returns the right names', () => {
        const names = response.body.currentPeriod.map((item) => item.name);
        expectSnapshot(names.sort()).toMatchInline(`
          Array [
            "elasticsearch",
            "opbeans-java",
            "postgresql",
            "redis",
          ]
        `);
      });

      it('returns the right service names', () => {
        const serviceNames = response.body.currentPeriod
          .map((item) => (item.type === 'service' ? item.serviceName : undefined))
          .filter(Boolean);

        expectSnapshot(serviceNames.sort()).toMatchInline(`
          Array [
            "opbeans-java",
          ]
        `);
      });

      it('returns the right latency values', () => {
        const latencyValues = sortBy(
          response.body.currentPeriod.map((item) => ({
            name: item.name,
            currentPeriodLatency: item.latency.value,
          })),
          'name'
        );

        expectSnapshot(latencyValues).toMatchInline(`
          Array [
            Object {
              "currentPeriodLatency": 2628.905,
              "name": "elasticsearch",
            },
            Object {
              "currentPeriodLatency": 27859.2857142857,
              "name": "opbeans-java",
            },
            Object {
              "currentPeriodLatency": 28580.1312997347,
              "name": "postgresql",
            },
            Object {
              "currentPeriodLatency": 1135.15508885299,
              "name": "redis",
            },
          ]
        `);
      });

      it('returns the right throughput values', () => {
        const throughputValues = sortBy(
          response.body.currentPeriod.map((item) => ({
            name: item.name,
            currentPeriodThroughput: item.throughput.value,
          })),
          'name'
        );

        expectSnapshot(throughputValues).toMatchInline(`
          Array [
            Object {
              "currentPeriodThroughput": 13.3333333333333,
              "name": "elasticsearch",
            },
            Object {
              "currentPeriodThroughput": 0.466666666666667,
              "name": "opbeans-java",
            },
            Object {
              "currentPeriodThroughput": 50.2666666666667,
              "name": "postgresql",
            },
            Object {
              "currentPeriodThroughput": 41.2666666666667,
              "name": "redis",
            },
          ]
        `);
      });

      it('returns the right impact values', () => {
        const impactValues = sortBy(
          response.body.currentPeriod.map((item) => ({
            name: item.name,
            currentPeriod: {
              impact: item.impact,
              latency: item.latency.value,
              throughput: item.throughput.value,
            },
          })),
          'name'
        );

        expectSnapshot(impactValues).toMatchInline(`
          Array [
            Object {
              "currentPeriod": Object {
                "impact": 1.54893576051104,
                "latency": 2628.905,
                "throughput": 13.3333333333333,
              },
              "name": "elasticsearch",
            },
            Object {
              "currentPeriod": Object {
                "impact": 0,
                "latency": 27859.2857142857,
                "throughput": 0.466666666666667,
              },
              "name": "opbeans-java",
            },
            Object {
              "currentPeriod": Object {
                "impact": 100,
                "latency": 28580.1312997347,
                "throughput": 50.2666666666667,
              },
              "name": "postgresql",
            },
            Object {
              "currentPeriod": Object {
                "impact": 2.37724265214801,
                "latency": 1135.15508885299,
                "throughput": 41.2666666666667,
              },
              "name": "redis",
            },
          ]
        `);
      });
    }
  );
}
