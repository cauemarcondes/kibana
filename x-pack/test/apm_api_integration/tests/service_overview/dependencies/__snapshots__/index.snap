// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APM API tests basic apm_8.0.0 Service overview dependencies when data is loaded returns at least one item 1`] = `
Object {
  "errorRate": Object {
    "timeseries": Array [
      Object {
        "x": 1607436720000,
        "y": 0,
      },
      Object {
        "x": 1607436780000,
        "y": 0,
      },
      Object {
        "x": 1607436840000,
        "y": 0,
      },
      Object {
        "x": 1607436900000,
        "y": 0,
      },
      Object {
        "x": 1607436960000,
        "y": 0,
      },
      Object {
        "x": 1607437020000,
        "y": 0,
      },
      Object {
        "x": 1607437080000,
        "y": 0,
      },
      Object {
        "x": 1607437140000,
        "y": 0,
      },
      Object {
        "x": 1607437200000,
        "y": 0,
      },
      Object {
        "x": 1607437260000,
        "y": 0,
      },
      Object {
        "x": 1607437320000,
        "y": 0,
      },
      Object {
        "x": 1607437380000,
        "y": 0,
      },
      Object {
        "x": 1607437440000,
        "y": 0,
      },
      Object {
        "x": 1607437500000,
        "y": 0,
      },
      Object {
        "x": 1607437560000,
        "y": 0,
      },
      Object {
        "x": 1607437620000,
        "y": 0,
      },
    ],
    "value": 0,
  },
  "impact": 2.37724265214801,
  "latency": Object {
    "timeseries": Array [
      Object {
        "x": 1607436720000,
        "y": 1344.68,
      },
      Object {
        "x": 1607436780000,
        "y": 731.439024390244,
      },
      Object {
        "x": 1607436840000,
        "y": 945.684210526316,
      },
      Object {
        "x": 1607436900000,
        "y": 1187.1914893617,
      },
      Object {
        "x": 1607436960000,
        "y": 884.705882352941,
      },
      Object {
        "x": 1607437020000,
        "y": 839.604651162791,
      },
      Object {
        "x": 1607437080000,
        "y": 1518.38636363636,
      },
      Object {
        "x": 1607437140000,
        "y": 930.9,
      },
      Object {
        "x": 1607437200000,
        "y": 868.230769230769,
      },
      Object {
        "x": 1607437260000,
        "y": 892.842105263158,
      },
      Object {
        "x": 1607437320000,
        "y": 995.021739130435,
      },
      Object {
        "x": 1607437380000,
        "y": 3405.75,
      },
      Object {
        "x": 1607437440000,
        "y": 976.102564102564,
      },
      Object {
        "x": 1607437500000,
        "y": 1167.9,
      },
      Object {
        "x": 1607437560000,
        "y": 839.463414634146,
      },
      Object {
        "x": 1607437620000,
        "y": 775.857142857143,
      },
    ],
    "value": 1135.15508885299,
  },
  "name": "redis",
  "spanSubtype": "redis",
  "spanType": "db",
  "throughput": Object {
    "timeseries": Array [
      Object {
        "x": 1607436720000,
        "y": 1.66666666666667,
      },
      Object {
        "x": 1607436780000,
        "y": 2.73333333333333,
      },
      Object {
        "x": 1607436840000,
        "y": 2.53333333333333,
      },
      Object {
        "x": 1607436900000,
        "y": 3.13333333333333,
      },
      Object {
        "x": 1607436960000,
        "y": 2.26666666666667,
      },
      Object {
        "x": 1607437020000,
        "y": 2.86666666666667,
      },
      Object {
        "x": 1607437080000,
        "y": 2.93333333333333,
      },
      Object {
        "x": 1607437140000,
        "y": 2.66666666666667,
      },
      Object {
        "x": 1607437200000,
        "y": 2.6,
      },
      Object {
        "x": 1607437260000,
        "y": 2.53333333333333,
      },
      Object {
        "x": 1607437320000,
        "y": 3.06666666666667,
      },
      Object {
        "x": 1607437380000,
        "y": 2.4,
      },
      Object {
        "x": 1607437440000,
        "y": 2.6,
      },
      Object {
        "x": 1607437500000,
        "y": 2.66666666666667,
      },
      Object {
        "x": 1607437560000,
        "y": 2.73333333333333,
      },
      Object {
        "x": 1607437620000,
        "y": 1.86666666666667,
      },
    ],
    "value": 41.2666666666667,
  },
  "type": "external",
}
`;

exports[`APM API tests basic no data Service overview dependencies when specific data is loaded returns opbeans-node as a dependency 1`] = `
Object {
  "agentName": "nodejs",
  "environment": "",
  "errorRate": Object {
    "timeseries": Array [
      Object {
        "x": 1607436720000,
        "y": 0.333333333333333,
      },
      Object {
        "x": 1607436780000,
        "y": null,
      },
      Object {
        "x": 1607436840000,
        "y": null,
      },
      Object {
        "x": 1607436900000,
        "y": null,
      },
      Object {
        "x": 1607436960000,
        "y": null,
      },
      Object {
        "x": 1607437020000,
        "y": null,
      },
      Object {
        "x": 1607437080000,
        "y": null,
      },
      Object {
        "x": 1607437140000,
        "y": null,
      },
      Object {
        "x": 1607437200000,
        "y": null,
      },
      Object {
        "x": 1607437260000,
        "y": null,
      },
      Object {
        "x": 1607437320000,
        "y": null,
      },
      Object {
        "x": 1607437380000,
        "y": null,
      },
      Object {
        "x": 1607437440000,
        "y": null,
      },
      Object {
        "x": 1607437500000,
        "y": null,
      },
      Object {
        "x": 1607437560000,
        "y": null,
      },
      Object {
        "x": 1607437620000,
        "y": 0,
      },
    ],
    "value": 0.25,
  },
  "impact": 100,
  "latency": Object {
    "timeseries": Array [
      Object {
        "x": 1607436720000,
        "y": 6.66666666666667,
      },
      Object {
        "x": 1607436780000,
        "y": null,
      },
      Object {
        "x": 1607436840000,
        "y": null,
      },
      Object {
        "x": 1607436900000,
        "y": null,
      },
      Object {
        "x": 1607436960000,
        "y": null,
      },
      Object {
        "x": 1607437020000,
        "y": null,
      },
      Object {
        "x": 1607437080000,
        "y": null,
      },
      Object {
        "x": 1607437140000,
        "y": null,
      },
      Object {
        "x": 1607437200000,
        "y": null,
      },
      Object {
        "x": 1607437260000,
        "y": null,
      },
      Object {
        "x": 1607437320000,
        "y": null,
      },
      Object {
        "x": 1607437380000,
        "y": null,
      },
      Object {
        "x": 1607437440000,
        "y": null,
      },
      Object {
        "x": 1607437500000,
        "y": null,
      },
      Object {
        "x": 1607437560000,
        "y": null,
      },
      Object {
        "x": 1607437620000,
        "y": 1,
      },
    ],
    "value": 5.25,
  },
  "name": "opbeans-node",
  "serviceName": "opbeans-node",
  "throughput": Object {
    "timeseries": Array [
      Object {
        "x": 1607436720000,
        "y": 0.2,
      },
      Object {
        "x": 1607436780000,
        "y": null,
      },
      Object {
        "x": 1607436840000,
        "y": null,
      },
      Object {
        "x": 1607436900000,
        "y": null,
      },
      Object {
        "x": 1607436960000,
        "y": null,
      },
      Object {
        "x": 1607437020000,
        "y": null,
      },
      Object {
        "x": 1607437080000,
        "y": null,
      },
      Object {
        "x": 1607437140000,
        "y": null,
      },
      Object {
        "x": 1607437200000,
        "y": null,
      },
      Object {
        "x": 1607437260000,
        "y": null,
      },
      Object {
        "x": 1607437320000,
        "y": null,
      },
      Object {
        "x": 1607437380000,
        "y": null,
      },
      Object {
        "x": 1607437440000,
        "y": null,
      },
      Object {
        "x": 1607437500000,
        "y": null,
      },
      Object {
        "x": 1607437560000,
        "y": null,
      },
      Object {
        "x": 1607437620000,
        "y": 0.0666666666666667,
      },
    ],
    "value": 0.266666666666667,
  },
  "type": "service",
}
`;
